### PR TITLE
server/admin: Encode protocol buffers in accordance with the spec

### PR DESCRIPTION
**Pull request by tamird**
@mrtracy this should encode protos correctly; can you verify?

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/6587)

### CODE REVIEW

**Comment by tamird** [comment]
Gah, depends on https://github.com/gengo/grpc-gateway/pull/152

File: util/protoutil/jsonpb_marshal.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+		}
+		return buf.Bytes(), nil
+	}
+	return nil, fmt.Errorf("unexpected type %T does not implement %s", v, typeProtoMessage)
+}
+
+// Unmarshal unmarshals JSON "data" into "v"
+func (j *JSONPb) Unmarshal(data []byte, v interface{}) error {
+	if pb, ok := v.(proto.Message); ok {
+		return proto.Unmarshal(data, pb)
```
Comment (awalterschulze) [inline util/protoutil/jsonpb_marshal.go]: jsonpb.Unmarshal ?

Comment (tamird) [inline util/protoutil/jsonpb_marshal.go]: fixed, thanks

**Comment by bdarnell** [comment]
:lgtm:

<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> #### server/admin: Encode protocol buffers in accordance with the spec
> 
> @mrtracy this should encode protos correctly; can you verify?
> 
> 
> </details>


---

Reviewed 1 of 4 files at r2, 12 of 12 files at r5.
Review status: all files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---

*[GLOCKFILE, line 22 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6587#-KHzl9-DEz76q_qSo-3n:-KHzl9-EimkQyozVXGT3:1902161913) ([raw file](https://github.com/cockroachdb/cockroach/blob/23be2b5ebea3615bab6038903c7f834aac4c79ee/GLOCKFILE#L22)):*
> ```
> github.com/agtorre/gocolorize f42b554bf7f006936130c9bb4f971afd2d87f671
> github.com/biogo/store 913427a1d5e89604e50ea1db0f28f34966d61602
> github.com/chzyer/readline 92c174e5fb5590f1ddbabd5963ffc799f2323b10
> ```

Please put mass glockfile updates in their own commit (except for changes that are required by the dep updates)

---

*[util/http_test.go, line 118 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6587#-KHzp0wZ7H9UDxZnRmvs:-KHzp0wZ7H9UDxZnRmvt:975378093) ([raw file](https://github.com/cockroachdb/cockroach/blob/23be2b5ebea3615bab6038903c7f834aac4c79ee/util/http_test.go#L118)):*
> ```Go
> }
> 
> type readCloser struct {
> ```

This exists as `io/ioutil.NopCloser`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6587#-:-KHzptFwqRXzbs3UkbPR:1417088721)*

**Comment by tamird** [comment]
Review status: 1 of 13 files reviewed at latest revision, 3 unresolved discussions, some commit checks pending.

---

*[GLOCKFILE, line 22 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6587#-KHzl9-DEz76q_qSo-3n:-KHzrA3xD8e4s92BcDCD:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/23be2b5ebea3615bab6038903c7f834aac4c79ee/GLOCKFILE#L22)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary>
> Please put mass glockfile updates in their own commit (except for changes that are required by the dep updates)</details>
Done.

---

*[util/http_test.go, line 118 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6587#-KHzp0wZ7H9UDxZnRmvs:-KHzqW6Frk2NmUEAGjhJ:-1580500651) ([raw file](https://github.com/cockroachdb/cockroach/blob/23be2b5ebea3615bab6038903c7f834aac4c79ee/util/http_test.go#L118)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary>
> This exists as `io/ioutil.NopCloser`</details>
Nice, I was hoping it was so. Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6587)*

**Comment by mrtracy** [comment]
:lgtm:

<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary>
> :lgtm:</details>


---

Review status: 1 of 13 files reviewed at latest revision, 3 unresolved discussions, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6587#-:-KI-37vhuXzE6Os95-eR:1417088721)*

**Pull request merged**